client/components/notice/index.tsx

```diff
@@ -0,0 +1,146 @@
+/**
+ * External dependencies
+ */
+
+import React, { useEffect, FunctionComponent, useRef } from 'react';
+import classnames from 'classnames';
+import { localize } from 'i18n-calypso';
+import Gridicon from 'calypso/components/gridicon';
+
+/**
+ * Internal dependencies
+ */
+import { TimerHandle } from 'calypso/types';
+/**
+ * Style dependencies
+ */
+import './style.scss';
+
+/**
+ * Module constants
+ */
+const GRIDICONS_WITH_DROP = [
+	'add',
+	'cross-circle',
+	'ellipsis-circle',
+	'help',
+	'info',
+	'notice',
+	'pause',
+	'play',
+	'spam',
+];
+
+/**
+ * Constants
+ */
+enum Status {
+	Error = 'is-error',
+	Info = 'is-info',
+	Success = 'is-success',
+	Warning = 'is-warning',
+	Plain = 'is-plain',
+}
+
+interface Props {
+	className?: string;
+	duration?: number;
+	onDismissClick: () => void;
+	translate: ( x: string ) => string;
+	children: React.ReactNode;
+	status?: Status;
+	icon?: string | null;
+	isCompact?: boolean;
+	isLoading?: boolean;
+	showDismiss?: boolean;
+	text?: string | React.ReactNode;
+}
+
+const getIcon = ( status: Status ): string => {
+	let icon: string;
+
+	switch ( status ) {
+		case Status.Info:
+			icon = 'info';
+			break;
+		case Status.Success:
+			icon = 'checkmark';
+			break;
+		case Status.Error:
+			icon = 'notice';
+			break;
+		case Status.Warning:
+			icon = 'notice';
+			break;
+		default:
+			icon = 'info';
+			break;
+	}
+
+	return icon;
+};
+
+const Notice: FunctionComponent< Props > = ( {
+	className = '',
```

Comment:
It's so weird to pass an empty string as default here. Won't an `undefined` work the same way when evaluated by `classNames` below?

Reply:
Good point, https://github.com/Automattic/wp-calypso/pull/51686/commits/48097a9e14d33a5dcba11bc748501f651fc8b843